bindinfo/handle.go

```diff
@@ -838,57 +837,111 @@ func GenerateBindSQL(ctx context.Context, stmtNode ast.StmtNode, planHint string
 			return ""
 		}
 	}
-	// We need to evolve plan based on the current sql, not the original sql which may have different parameters.
-	// So here we would remove the hint and inject the current best plan hint.
-	hint.BindHint(stmtNode, &hint.HintsSet{})
-	bindSQL := utilparser.RestoreWithDefaultDB(stmtNode, defaultDB, "")
-	if bindSQL == "" {
-		return ""
-	}
+	// We can place hint at update, delete, select stmt block,there are three combinations follows:
+	// updateStmt-selectStmt,deleteStmt-selectStmt,insertStmt-selectStmt,
+	// so when these are present，we need a way to prevent the hint of wrong placement,
+	// eg `update /*+ use_index(@upd_1 t1 idx_b) t1 set b = 1 where  (a in (select /*+use_index(@sel_1 t2  idx_ba)*/ a from t2 where b = 1) )`
+	// if we only order hint by parse offset from qb_name then @upd_1 and @sel_1 will be same offset,
+	// it's incorrect,so we need a way to deal this，so introduce the startBlockOffset，
+	// the startBlockOffset is the start of the offset to place hint.now if we get @upd_1 and @sel_1,we will set startBlockOffset=0,
+	// because ParseOffsetFromBlockName only parse qbName which Prefix with  'sel_',
+	// so upd_1 will be parsed to startBlockOffset that is 0,the @sel_1 will be parse to 1,it right result for us
```

Comment:
```suggestion
	// We can place hints at update, delete, select stmt block, there are three combinations follows:
	// updateStmt-selectStmt, deleteStmt-selectStmt, insertStmt-selectStmt.
	// So when they appeared，we need a way to prevent the hints of the wrong place.
	// eg `update /*+ use_index(@upd_1 t1 idx_b) t1 set b = 1 where  (a in (select /*+use_index(@sel_1 t2  idx_ba)*/ a from t2 where b = 1) )`
	// If we only order hints by parsing offset from qb_name, then @upd_1 and @sel_1 will be the same offset.
	// It's incorrect, so we need a way to deal this. So we introduce the startBlockOffset.
	// The startBlockOffset is the start of the offset to place hints. Now if we get @upd_1 and @sel_1, we will set startBlockOffset=0.
	// Because ParseOffsetFromBlockName only parse qbName which Prefix with  'sel_',
	// so upd_1 will be parsed to startBlockOffset that is 0, the @sel_1 will be parse to 1. It's the right result for us.
```

Reply:
Format the comments.

Reply:
done